src/auth/auth-events/signout.ts

```diff
@@ -0,0 +1,108 @@
+import { sessionTimedOutErrorMessage } from 'src/auth/auth-errors';
+import { logoutCallbackLinkName } from 'src/auth/Logout';
+import { removeUserFromLocalState } from 'src/auth/oidc-broker';
+import { Ajax } from 'src/libs/ajax';
+import { getSessionStorage } from 'src/libs/browser-storage';
+import Events, { MetricsEventName } from 'src/libs/events';
+import * as Nav from 'src/libs/nav';
+import { notify, sessionTimeoutProps } from 'src/libs/notifications';
+import {
+  authStore,
+  azureCookieReadyStore,
+  cookieReadyStore,
+  MetricState,
+  metricStore,
+  oidcStore,
+  TokenMetadata,
+  userStore,
+} from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { DEFAULT, getTimestampMetricLabel, switchCase } from 'src/libs/utils';
+
+export type SignOutCause =
+  | 'requested'
+  | 'disabled'
+  | 'declinedTos'
+  | 'expiredRefreshToken'
+  | 'errorRefreshingAuthToken'
+  | 'idleStatusMonitor'
+  | 'unspecified';
+
+export const signOut = (cause: SignOutCause = 'unspecified'): void => {
+  // TODO: invalidate runtime cookies https://broadworkbench.atlassian.net/browse/IA-3498
+  sendSignOutMetrics(cause);
```

Comment:
As long as we're here, since this has caused problems before, would it be worth adding a comment that `sendSignOutMetrics` should _not_ be awaited, since it will fail if the sign out is due to session expiration?